### ISSUE TITLE
[BUG] `no-magic-number` false-negatives on list index access.

### ISSUE THREAD

**Issue by solid-yuriiprykhodko** [dart-code-checker/dart-code-metrics#695]
* **Dart code metrics version:** 4.10.1
* **Dart sdk version:** Dart SDK version: 2.15.1 (stable) (Tue Dec 14 13:32:21 2021 +0100) on "macos_x64"

**Please show your full configuration:**
Full repro project: https://github.com/solid-yuriiprykhodko/metrics_cc_repro/tree/magic-number-index-access
<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  rules:
    - no-magic-number
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
  List<num> l = List.generate(10, (i) => i);

  // write to mutable variable -> no lint
  num v = l[5];

  // write to final variable -> no lint
  final f = l[5];

  // pass to a function -> lint
  print(l[5]);

  Map<int, String> m = {
    1: '',
    2: '',
    3: '',
  };

  // write to mutable variable -> no lint
  String? mv = m[2];

  // write to final variable -> no lint
  final mf = m[2];

  // pass to a function -> lint
  print(m[2]);
```

**What did you expect to happen?**
`no-magic-number` lint triggered for all literal `int` index or map key access, possibly except the `final` variable case.

**What actually happened?**
`no-magic-number` lint triggered only outside variable assignment.

**Are you willing to submit a pull request to fix this bug?**
No.

**Comment by incendial** [comment]
Available in 4.14.0 🚀
